pkg/csi/service/vanilla/controller.go

```diff
@@ -596,77 +597,148 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 		ContentSourceSnapshotID: contentSourceSnapshotID,
 	}
 
+	// Check if vCenter task for this volume is already registered as part of
```

Comment:
Does this logic not apply to file volume creation?

Reply:
Yes, we need to make changes for file volume as well. I will make that change through a separate PR. This PR is dedicated to block volume changes.